### ISSUE TITLE
capture authorized payment by changing order status (or programmatically) (PCP-235)

### ISSUE THREAD

**Issue by Sirvijver** [woocommerce/woocommerce-paypal-payments#587]
**Description**
When a payment was authorized, the WooCommerce order will be set to On-Hold until the payment has been captured.
The payment can be captured through the WooCommerce order actions.

PayPal Checkout Gateway and Stripe for example will automatically capture an authorized payment when the order status is changed from `On-Hold` to `Processing`.

Unlike PayPal Checkout Gateway and Stripe, PayPal Payments will not automatically capture the payment when the order status is changed from `On-Hold` to `Processing`:
![image](https://user-images.githubusercontent.com/78153894/163161723-5dcf5e22-ea55-4f8e-9129-119e98e4c940.png)

I propose adding functionality that will automatically capture authorized payments when the order status is changed to `Processing`.
This will be beneficial when external ERP or fraud prevention systems are used as they usually capture authorized payments by changing the order status.

We also had some 3dp developers reach out because they want to provide their users the option to capture a payment from their account page. But the approach they used with the previous Checkout plugin does not work with PCP.

They were looking for something like this: `do_action('woocommerce_order_action_ppcp_authorize_order', $wc_order)`  .

Adding something like `ppcp_capture( $order_id )` and function like `ppcp_is_authorized( $order_id )` may benefit other developers too.

**Comment by ifnull** [comment]
Relates to: https://github.com/woocommerce/woocommerce-paypal-payments/issues/524

@Sirvijver 

We were using the code below to accomplish just this but it recently started to fail. In fact, for these orders, we can no longer capture from the order actions. We now receive the error `Could not retrieve information. Try again later`. 

```
use WooCommerce\PayPalCommerce\WcGateway\Gateway\PayPalGateway;

class Payments{

	/**
	 * Constructor
	 */

	public function __construct() {
		add_action( 'woocommerce_order_status_completed', [ $this, 'capture_payment' ] );
	}

	/**
	 * Capture payment.
	 *
	 * @param int $order_id
	 *
	 * @return void
	 */

	public function capture_payment( $order_id ) {

		if ( ! class_exists( '\WooCommerce\PayPalCommerce\WcGateway\Gateway\PayPalGateway' ) ) {
			return;
		}

		if ( $order = wc_get_order( $order_id ) ) {

			// Check to see if we should process this payment through WooCommerce PayPal Payments plugin.
			if ( $transaction_id = $order->get_meta( PayPalGateway::ORDER_ID_META_KEY ) ) {

				if ( ! empty( $transaction_id ) ) {
					do_action( 'woocommerce_order_action_ppcp_authorize_order', $order );
				}

			}

		}

	}

}
```

**Comment by Sirvijver** [comment]
Hello @ifnull ,

This issue (Could not retrieve information. Try again later) should have been resolved with the release of 1.9.0
We still have this feature request on our list, but we can still not give you a date when this will be integrated into the plugin.
